### ISSUE TITLE
ip_and_mask variable in os10_vlan try to create ip on physical interface instead on a vlan

### ISSUE THREAD

**Issue by woyteck1** [ansible-collections/dellemc.os10#42]
##### SUMMARY
ip_and_mask variable in os10_vlan try to create ip on physical interface instead on a vlan as soon as whichever physical interface is added to tagged or untagged members
os10 has an option to have an IP directly in a vlan, so it is not associated with a physical interface. 
However as soon as tagged or untagged members are listed, this role insists on adding the IP to a physical interface.
If the interface is in a switchport mode, an IP cannot be added to the interface.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
os10_vlan

##### ANSIBLE VERSION

```paste below
ansible 2.10.1
  config file = /home/wojciechk/git/auto/ansible.cfg
  configured module search path = [u'/home/wojciechk/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/wojciechk/git/auto/.venv/local/lib/python2.7/site-packages/ansible
  executable location = /home/wojciechk/git/auto/.venv/bin/ansible
  python version = 2.7.12 (default, Nov 12 2018, 14:36:49) [GCC 5.4.0 20160609]

```

##### CONFIGURATION

```paste below
ANSIBLE_COW_SELECTION(/home/wojciechk/git/auto/ansible.cfg) = stegosaurus
ANSIBLE_FORCE_COLOR(/home/wojciechk/git/auto/ansible.cfg) = True
ANSIBLE_NOCOWS(/home/wojciechk/git/auto/ansible.cfg) = True
DEFAULT_FORKS(/home/wojciechk/git/auto/ansible.cfg) = 20
DEFAULT_LOG_PATH(/home/wojciechk/git/auto/ansible.cfg) = /home/REDACTED/git/auto/log_ansible.txt
DEFAULT_MANAGED_STR(/home/wojciechk/git/auto/ansible.cfg) = This file is being managed with Ansible. Do NOT update it manually.
DEFAULT_ROLES_PATH(/home/wojciechk/git/auto/ansible.cfg) = [u'/home/REDACTED/git/auto/galaxy-roles', u'/home/wojciechk/git/auto/roles', u'/home/REDACTED/git/auto/galaxy-role
DEFAULT_TRANSPORT(/home/wojciechk/git/auto/ansible.cfg) = ssh
DISPLAY_SKIPPED_HOSTS(/home/wojciechk/git/auto/ansible.cfg) = False
HOST_KEY_CHECKING(/home/wojciechk/git/auto/ansible.cfg) = False
INTERPRETER_PYTHON(/home/wojciechk/git/auto/ansible.cfg) = /usr/bin/python3
INVENTORY_ENABLED(/home/wojciechk/git/auto/ansible.cfg) = [u'auto', u'vmware_vm_inventory', u'yaml', u'host_list']

```

##### STEPS TO REPRODUCE

Have a port in switchport mode.
Create a vlan with ip added and said port being an untagged (or tagged) member.

```yaml
    vlan 2254:
      ip_and_mask: "172.21.254.254/30"
      tagged_members:
          - port: ethernet 1/1/1
            state: present


```
On switch:

interface ethernet1/1/1
 no shutdown
 switchport mode trunk
 switchport access vlan 2254
 switchport trunk allowed vlan 2000-2001
 flowcontrol receive on

##### EXPECTED RESULTS

Expected result is a vlan with ip address configured.
`
interface vlan2254

 no shutdown

 ip address 172.21.254.254/30
`

##### ACTUAL RESULTS

```paste below
_collections/dellemc/os10/roles/os10_vlan/tasks/main.yml:11
fatal: [lon1sw1]: FAILED! => {
    "changed": false,
    "command": "ip address 172.21.254.254/30",
    "invocation": {
        "module_args": {
            "after": null,
            "backup": false,
            "backup_options": null,
            "before": null,
            "config": null,
            "lines": null,
            "match": "line",
            "parents": null,
            "provider": null,
            "replace": "line",
            "save": false,
            "src": "\n\ninterface vlan2255\n description \"Inter-Switch network\"\n\n\n ip address 172.21.255.1/26\n\n\n\n\n\ninterface vlan2254\n description \"lon1sw1-to-Meraki link\"\n\ninterface ethernet 1/1/1\n switchport mode trunk\n switchport trunk allowed vlan 2254\n\n ip address 172.21.254.254/30\n\n\n\ndefault vlan-id 1\n\n\ninterface vlan2000\n description \"Management VLAN\"\n\n\n\n\n\n\n\ninterface vlan2001\n description \"AML machines\"\ninterface ethernet 1/1/3\n switchport access vlan 2001\ninterface ethernet 1/1/4\n switchport access vlan 2001\n\ninterface ethernet 1/1/1\n switchport mode trunk\n switchport trunk allowed vlan 2001\n\n\n\n\n",
            "update": "merge"
        }
    },
    "msg": "ip address 172.21.254.254/30\r\n% Error: Interface ethernet1/1/1, IP address cannot exist with L2 modes.\r\nlon1sw1(conf-if-eth1/1/1)# ",
    "rc": -32603
}


```

**Comment by woyteck1** [comment]
One more question - I see no option to enable "mode L3" on interface vlan, and on interface ethernet.
Any plans in adding it?

Thanks

**Comment by gunasekaran-t** [comment]
Fix is available in collection version 1.0.2
